src/kokkos/plugin-BeamSpotProducer/kokkos/BeamSpotToKokkos.cc

```diff
@@ -0,0 +1,34 @@
+#include "KokkosDataFormats/BeamSpotKokkos.h"
+
+#include "Framework/EDProducer.h"
+#include "Framework/Event.h"
+#include "Framework/EventSetup.h"
+#include "Framework/PluginFactory.h"
+
+namespace KOKKOS_NAMESPACE {
+  class BeamSpotToKokkos : public edm::EDProducer {
+  public:
+    explicit BeamSpotToKokkos(edm::ProductRegistry& reg);
+    ~BeamSpotToKokkos() override = default;
+
+    void produce(edm::Event& iEvent, const edm::EventSetup& iSetup) override;
+
+  private:
+    edm::EDPutTokenT<BeamSpotKokkos<KokkosExecSpace>> bsPutToken_;
+
+    typename Kokkos::View<BeamSpotKokkos<KokkosExecSpace>::Data, KokkosExecSpace>::HostMirror bsHost;
+  };
+
+  BeamSpotToKokkos::BeamSpotToKokkos(edm::ProductRegistry& reg)
+      : bsPutToken_{reg.produces<BeamSpotKokkos<KokkosExecSpace>>()},
+        bsHost{"bsHost"} {}
+
+  void BeamSpotToKokkos::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+    auto bsRaw = iSetup.get<BeamSpotKokkos<KokkosExecSpace>::Data>();
```

Comment:
```suggestion
    auto const& bsRaw = iSetup.get<BeamSpotKokkos<KokkosExecSpace>::Data>();
```
The `BeamSpotKokkos<KokkosExecSpace>::Data` looks weird but is technically correct. Maybe it would actually be clearer to rename `BeamSpotKokkos<...>::Data` to e.g. `BeamSpotPOD`, and use `Kokkos::View<BeamSpotPOD, KokkosExecSpace>` and use that everywhere.